eng/common/scripts/ChangeLog-Operations.ps1

```diff
@@ -362,4 +321,72 @@ function  Get-LatestReleaseDateFromChangeLog
   $changeLogEntries = Get-ChangeLogEntries -ChangeLogLocation $ChangeLogLocation
   $latestVersion = $changeLogEntries[0].ReleaseStatus.Trim("()")
   return ($latestVersion -as [DateTime])
+}
+
+function Confirm-LikeForRelease {
+  param (
+    [Parameter(Mandatory = $true)]
+    $changeLogEntry,
+    [Parameter(Mandatory = $true)]
+    $changeLogFileDateHistory
+  )
+
+  $isValid = $true
+  if ($changeLogEntry.ReleaseStatus -eq $CHANGELOG_UNRELEASED_STATUS) {
+    LogError "Entry has no release date set. Please ensure to set a release date with format '$CHANGELOG_DATE_FORMAT'. See https://aka.ms/azsdk/guideline/changelogs for more info."
+    $isValid = $false
+  }
+  else {
+    $status = $changeLogEntry.ReleaseStatus.Trim().Trim("()")
+    try {
+      $releaseDate = [DateTime]$status
+      if ($status -ne ($releaseDate.ToString($CHANGELOG_DATE_FORMAT)))
+      {
+        LogError "Date must be in the format $($CHANGELOG_DATE_FORMAT). See https://aka.ms/azsdk/guideline/changelogs for more info."
+        $isValid = $false
+      }
+
+      $dateHistory = $changeLogFileDateHistory | ForEach-Object { $_.Trim("()") } | Sort-Object -Descending
+      if (@($dateHistory)[0] -ne $status)
+      {
+        LogError "Invalid date [ $status ]. The date for the changelog being released must be the latest in the file."
+        $isValid = $false
+      }
+    }
+    catch {
+        LogError "Invalid date [ $status ] passed as status for Version [$($changeLogEntry.ReleaseVersion)]. See https://aka.ms/azsdk/guideline/changelogs for more info."
+        $isValid = $false
+    }
+  }
+
+  if ([System.String]::IsNullOrWhiteSpace($changeLogEntry.ReleaseContent)) {
+    LogError "Entry has no content. Please ensure to provide some content of what changed in this version. See https://aka.ms/azsdk/guideline/changelogs for more info."
+    $isValid = $false
+  }
+
+  $foundRecommendedSection = $false
+  $emptySections = @()
+  foreach ($key in $changeLogEntry.Sections.Keys)
+  {
+    $sectionContent = $changeLogEntry.Sections[$key]
+    if ([System.String]::IsNullOrWhiteSpace(($sectionContent | Out-String)))
+    {
+      $emptySections += $key
+    }
+    if ($RecommendedSectionHeaders -contains $key)
+    {
+      $foundRecommendedSection = $true
+    }
+  }
+  if ($emptySections.Count -gt 0)
+  {
+    LogError "The changelog entry has the following sections with no content ($($emptySections -join ', ')). Please ensure to either remove the empty sections or add content to the section."
+    $isValid = $false
+  }
+  if (!$foundRecommendedSection)
+  {
+    LogWarning "The changelog entry did not contain any of the recommended sections ($($RecommendedSectionHeaders -join ', ')), please add at least one. See https://aka.ms/azsdk/guideline/changelogs for more info."
+  }
+  Write-Host "Changelog validation failed. Please fix errors above and try again."
```

Comment:
This message doesn't seem correct.

Reply:
This is because I am reporting all the various errors that occur before exiting the validation process. Previously we just exited immediately an error occurs

Reply:
Shouldn't this message be removed?